.github/workflows/release-candidate.yml

```diff
@@ -0,0 +1,107 @@
+name: Release Candidate
+
+on:
+  # Manual Trigger GH CLI -> gh workflow run release-candidate.yml --ref <branch-name>
+  workflow_dispatch:
+
+jobs:
+  run-python-tests:
+    uses: ./.github/workflows/python-versions.yml
+    with:
+      ref: ${{ github.ref_name }}
+
+  run-javascript-tests:
+    uses: ./.github/workflows/js-tests.yml
+    with:
+      ref: ${{ github.ref_name }}
+
+  run-py-prod-deps-smoke-test:
+    uses: ./.github/workflows/py-prod-deps-smoke-test.yml
+    with:
+      ref: ${{ github.ref_name }}
+      
+  run-cypress-tests:
+    uses: ./.github/workflows/cypress.yml
+    with:
+      ref: ${{ github.ref_name }}
+      
+  build-release-candidate:
+    runs-on: ubuntu-latest
+
+    needs: [run-python-tests, run-javascript-tests, run-py-prod-deps-smoke-test, run-cypress-tests]
+
+    defaults:
+      run:
+        shell: bash -ileo pipefail {0}
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with: 
+          persist-credentials: false
+          submodules: 'recursive'
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Setup virtual env
+        uses: ./.github/actions/make_init
+      - name: Set Environment Variables
+        uses: ./.github/actions/branch
+        with:
+          branch: ${{ github.ref_name }}
+      - name: Set desired version from branch name
+        env:
+          BRANCH: ${{ env.BRANCH }}
+        run: |
+          echo "DESIRED_VERSION=$(echo "${BRANCH}" | tr -d "release/")" >> $GITHUB_ENV
```

Comment:
nit: I'd recommend picking one CLI-based text manipulation tool to do all of the substitutions/deletions here just so that the reader doesn't have to remind themselves of what several of them do. All of the string manipulation being done here is simple enough that I doubt the differences between them will ever be meaningful